sdk/src/credential_provider.rs

```diff
@@ -291,6 +310,15 @@ mod tests {
             credential_provider.control_endpoint,
             "https://control plane endpoint"
         );
+        assert_eq!(
+            "https://data plane endpoint",
+            credential_provider.token_endpoint
+        );
+
+        assert_eq!(
```

Comment:
I followed the existing pattern here, but this doesn't seem correct. Are the token and storage endpoints supposed to resolve to the same thing as the cache endpoint?

Reply:
let's just remove legacy jwt support

Reply:
I'll make a separate pr for that.

Reply:
https://github.com/momentohq/client-sdk-rust/pull/366